rules/linux/execution_shared_memory_executable.toml

```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2022/05/10"
+maturity = "production"
+updated_date = "2022/05/10"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of a binary in Linux shared memory directories, /dev/shm or /run/shm. Adversaries may attempt to
+launch a malicious executable from these temporary directories during initial access to a host.
+"""
+false_positives = [
+    """
+    Directories /dev/shm and /run/shm are temporary file storage directories in Linux. They are intended to appear as a
+    mounted file system, but uses virtual memory instead of a persistent storage device and thus are used for mounting
+    file systems in legitimate purposes.
+    """,
+]
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Binary Executed from Shared Memory Directory"
+references = [
+    "https://linuxsecurity.com/features/fileless-malware-on-linux",
+    "https://twitter.com/GossiTheDog/status/1522964028284411907",
+]
+risk_score = 43
+rule_id = "3f3f9fe2-d095-11ec-95dc-f661ea17fbce"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Execution"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type == "start" and event.action == "exec" and process.executable : ("/dev/shm/*","/run/shm/*")
```

Comment:
Can we look at linux_process_started_in_temp_directory.toml.
This can be combined or the rule scope can be expanded to what is available and make it more optimized. 

With prior experience having unique rules in 8.2 for each shell evasion of a process, combining them to a single rule this fits the perfect category for the same.

The rule looks legacy and does not have any mitre attack vectors, with this effort the rule can be improved as well in terms of detection factors.

Reply:
execution from temp is usually noisy, in general if a new logic has a different noise level than existing logic then better to not merge them in one rule since a user may decide to disable it (to reduce noise) and impacts the new logic (which normally not noisy).